virtual_rainforest/core/model.py

```diff
@@ -135,10 +137,21 @@ def __init_subclass__(cls, model_name: str):
     def setup(self) -> None:
         """Function to use input data to set up the model."""
 
+    def start_model_timing(self, start_time: datetime64) -> None:
+        """Set of model timing by setting up initial last update time."""
+
+        if hasattr(self, "last_update"):
+            log_and_raise(
+                "Model timing was already set up, it should not be setup again!",
+                ImproperFunctionCall,
+            )
+        else:
+            self.last_update = start_time
+
```

Comment:
Just make `start_time` part of the `__init__` arguments? No need to modularise this into a method and then have to guard against improper use.

Reply:
Yeah this is probably sensible

Reply:
I agree. Creating an attribute on the fly, when it is something you can provide directly in `__init__`, does not make much sense.